src/main/kotlin/dev/arbjerg/ukulele/audio/Player.kt

```diff
@@ -130,9 +136,14 @@ class Player(val beans: Beans, guildProperties: GuildProperties) : AudioEventAda
     }
 
     override fun onTrackEnd(player: AudioPlayer, track: AudioTrack, endReason: AudioTrackEndReason) {
-        if (isRepeating && endReason.mayStartNext) {
-            queue.add(track.makeClone())
+        if (endReason.mayStartNext) {
+            if (repeatOne) {
+                queue.addFirst(track.makeClone())
+            } else if (isLooping) {
+                queue.add(track.makeClone())
+            }
```

Comment:
If repeat (one) is enabled, add the same track to the top of the queue.  Else, if looping is enabled, add the track to the end of the queue.